indico/modules/events/sessions/forms.py

```diff
@@ -100,3 +102,24 @@ def session_fields(self):
     @property
     def block_fields(self):
         return [field_name for field_name in self._fields if field_name.startswith('block_')]
+
+
+class SessionTypeForm(IndicoForm):
+    """Form to create or edit a SessionType"""
+
+    name = StringField(_("Name"), [DataRequired()])
+    is_poster = BooleanField(_("Poster"), widget=SwitchWidget(),
+                             description=_("If selected, this session type will appear in timetable "
+                                           "as a poster session."))
+
+    def __init__(self, *args, **kwargs):
+        self.event = kwargs.pop('event')
+        self.session_type = kwargs.get('obj')
+        super(SessionTypeForm, self).__init__(*args, **kwargs)
+
+    def validate_name(self, field):
+        query = self.event.session_types.filter(db.func.lower(SessionType.name) == field.data.lower())
```

Comment:
If you get rid of the lazy=dynamic of the relationship, you need to use this instead:

    query = SessionType.query.with_parent(self.event).filter(...)